### PR TITLE
applab: add get/setSharedKey blocks. add hover text for all storage blocks.

### PR DESCRIPTION
**Pull request by davidsbailey**

### CODE REVIEW

**Comment by cpirich** [comment]
hover text will need to move into dropletConfig.js given my recent change

File: apps/src/applab/levels.js [hunk 1]
```diff
@@ -71,10 +71,12 @@ levels.ec_simple = {
...
     {'func': 'canvasPutImageData', 'title': 'Set the ImageData for a rectangle within a canvas with x, y as the top left coordinates', 'category': 'Canvas', 'params': ["'id'", "imageData", "0", "0"] },
     {'func': 'canvasClear', 'title': 'Clear all data on a canvas', 'category': 'Canvas', 'params': ["'id'"] },
-    {'func': 'createSharedRecord', 'category': 'Storage', 'params': ["{tableName:'abc', name:'Alice', age:7, male:false}", "function() {\n  \n}"] },
-    {'func': 'readSharedRecords', 'category': 'Storage', 'params': ["{tableName: 'abc'}", "function(records) {\n  for (var i =0; i < records.length; i++) {\n    createHtmlBlock('id', records[i].id + ': ' + records[i].name);\n  }\n}"] },
-    {'func': 'updateSharedRecord', 'category': 'Storage', 'params': ["{tableName:'abc', id: 1, name:'Bob', age:8, male:true}", "function() {\n  \n}"] },
-    {'func': 'deleteSharedRecord', 'category': 'Storage', 'params': ["{tableName:'abc', id: 1}", "function() {\n  \n}"] },
+    {'func': 'setSharedKey', 'title': 'Reads the value associated with the key, shared with everyone who uses the app.', 'category': 'Storage', 'params': ["'key'", "'value'", "function () {\n  \n}"] },
```
Comment (cpirich) [inline apps/src/applab/levels.js]: I think these descriptions are flipped (get/set)

**Comment by cpirich** [comment]
LGTM

General question: what does this do? Is it a higher-level function that is simpler than using records?
Naming question: does it really get/set a key or get/set a value given a key? I was confused by the name, expecting it to do something different.

**Comment by davidsbailey** [comment]
Yes, it get/set are high-level functions that are simpler than using records. They use a separate piece of storage infrastructure which only stores key-value pairs. The idea is to introduce storage in very simple ways, e.g. high score for games, before introducing the complexity of records.

naming: you are right, it does not really get/set a key. would get/setValue be better? or getProperty?

**Comment by cpirich** [comment]
I think readSharedValue / writeSharedValue would be reasonably consistent
with your record APIs (or maybe read / update?)

I also think getSharedValue / setSharedValue would make sense.

What is the function signature when they are no longer shared? I don't
think we can just have global functions called getValue / setValue because
it won't be obvious that they are cloud storage functions - especially when
in JavaScript mode without the palette there to remind you.

**Comment by davidsbailey** [comment]
synced, merged, and addressed all comments. going with readSharedValue / writeSharedValue for now.

**Pull request merged**